### PR TITLE
Add search component and SVG icon

### PR DESCRIPTION
**Pull request by global-usman**
- Added Search component with debounce functionality
- Included SVG icon component
- Implemented example usage of Search component
- Created stories and tests for Search component and SVG icon

### CODE REVIEW

File: src/components/search/search.css
Comment (darnold001) [inline src/components/search/search.css:1]: Please rewrite in styled-component syntax

**Review by darnold001 — COMMENTED** [review]
Looks good! Had one comment pertaining to CSS

File: src/components/search/Search.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  padding-bottom: 9px;
+  padding-right: 8px;
+  border: 1px solid var(--neutral-200, #9ca3af);
+  background: var(--system-50, #fff);
+  color: var(--neutral-200, #9ca3af);
+  box-shadow: 0px 1px 2px 0px rgba(0, 0, 0, 0.05);
+
+  &:focus {
+    border: 1px solid var(--Global-Tokens-Primary-100, #1c99ce);
+    outline: 1px solid var(--Global-Tokens-Primary-100, #1c99ce);
```
Comment (zamanaf) [inline src/components/search/Search.tsx]: Can we use `--primary-100` here? As we are using these variables in the app

File: src/components/search/Search.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  }
+
+  &:hover {
+    border: 1px solid var(--Global-Tokens-Primary-100, #1c99ce);
```
Comment (zamanaf) [inline src/components/search/Search.tsx]: same here

File: src/components/search/Search.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+}) => {
+  const [searchTerm, setSearchTerm] = useState("");
+
+  useEffect(() => {
+    const debounceTimer = setTimeout(() => {
+      onChangeCallback(searchTerm);
+    }, debounceTime);
+
+    return () => clearTimeout(debounceTimer);
+  }, [searchTerm, onChangeCallback, debounceTime]);
```
Comment (zamanaf) [inline src/components/search/Search.tsx]:
```suggestion
  }, [searchTere, debounceTime]);
```

File: src/components/search/Search.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    border: 1px solid var(--Global-Tokens-Primary-100, #1c99ce);
+    outline: 1px solid var(--Global-Tokens-Primary-100, #1c99ce);
+  }
+
+  &:hover {
+    border: 1px solid var(--Global-Tokens-Primary-100, #1c99ce);
+  }
+`;
+
+const SearchIcon = styled(SvgIcon)`
```
Comment (zamanaf) [inline src/components/search/Search.tsx]: Can we name this to `StyledIcon`?
```suggestion
const StyledIcon = styled(SvgIcon)`
```

**Review by zamanaf — APPROVED** [review]
LGTM

**Pull request merged**